### PR TITLE
feature: implement accuracy degradation profile

### PR DESCRIPTION
**Pull request by andrelfnovaes**

### CODE REVIEW

File: src/holisticai/robustness/dataset_shift/_accuracy_degradation_profile.py [hunk 1]
```diff
@@ -0,0 +1,345 @@
...
+) -> tuple[pd.DataFrame, list[float]]:
+    """
+    Calculate accuracies by iteratively reducing the test set size and evaluating accuracy.
+
+    This function evaluates the accuracy of predictions as the size of the test set is gradually reduced.
+    For each reduction, it calculates the accuracy over a subset of the nearest neighbors in the test set.
+
+    Parameters:
+    ----------
+    X_test : np.ndarray
```
Comment (crismunoz) [inline src/holisticai/robustness/dataset_shift/_accuracy_degradation_profile.py:165]: can be X_test a dataframe?
in tha case can be matrix-like for dataframe
for pd.Series is array-like

File: src/holisticai/utils/plots/_plots.py [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+    plt.figure(figsize=(8, 6))
+    plt.scatter(X[:, 0], X[:, 1], c=y, cmap="viridis", s=50, edgecolor="k")
+
+    plt.xlabel("Feature 1")
+    plt.ylabel("Feature 2")
+    plt.title("Your 2D Dataset")
+    plt.show()
+
+
+def plot_highlight_test_set(X, y, test_indices):
```
Comment (crismunoz) [inline src/holisticai/utils/plots/_plots.py]: i think we can merge this function with plot_graph and use test_indices as optional parameter

File: src/holisticai/utils/plots/_plots.py [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+        The predicted labels for each sample.
+
+    Returns:
+    -------
+    None
+    """
+    import matplotlib.pyplot as plt
+
+    # Adjustable vertical offset
+    vertical_offset = 0.1
```
Comment (crismunoz) [inline src/holisticai/utils/plots/_plots.py]: must be optional parameter

**Review by crismunoz — CHANGES_REQUESTED** [review]
nice work. Please check the commens i think we can create nice visualizations

File: src/holisticai/robustness/dataset_shift/_accuracy_degradation_profile.py (hunk 1 above)
Comment (andrelfnovaes) [reply src/holisticai/robustness/dataset_shift/_accuracy_degradation_profile.py:165]: Done.

File: src/holisticai/utils/plots/_plots.py (hunk 2 above)
Comment (andrelfnovaes) [reply src/holisticai/utils/plots/_plots.py]: Done.

File: src/holisticai/utils/plots/_plots.py (hunk 3 above)
Comment (andrelfnovaes) [reply src/holisticai/utils/plots/_plots.py]: Done.

**Review by crismunoz — APPROVED** [review]
good!

**Pull request merged**